traitsui/tests/editors/test_text_editor.py

```diff
@@ -163,12 +163,13 @@ def test_custom_editor_init_and_dispose_no_auto_set(self):
         self.check_editor_init_and_dispose(style="custom", auto_set=False)
 
     def test_simple_auto_set_update_text(self):
-        foo = Foo()
+        foo = Foo(name="a")
         view = get_view(style="simple", auto_set=True)
         tester = UITester()
         with tester.create_ui(foo, dict(view=view)) as ui:
-            with self.assertTraitChanges(foo, "name", count=3):
+            with self.assertTraitChanges(foo, "name", count=4):
                 name_field = tester.find_by_name(ui, "name")
+                name_field.perform(command.KeyClick("Backspace"))
```

Comment:
On Windows, it goes through the branch where there is selected text... because the first input is given focus with all text selected. On OSX, the input is initialized such that it does not already have focus, so we go through the other branch where we are backspacing from the end of text. Since there is only one character, the overall effect is the same.

On second thought, that's perhaps too much complication for the purpose of this test. I will remove this change. The unit test for key click is good enough.